### PR TITLE
Convert Objective-C AmericanExpress module to Swift

### PR DESCRIPTION
**Pull request by jaxdesmarais**
### Summary of changes

- Removed BraintreeAmericanExpress/Public directory
- Converted .h and .m files with single .swift files
- Updated podspec path to point to the updated file path
- Remove `BTAmericanExpressErrorDomain` global constant
- Remove `BTAmericanExpressErrorType`
- Make `BTAmericanExpressRewardsBalance` initializer private
- Created `BTAmericanExpressError.swift` that houses the domain, error code, and localized description of all the possible errors

### Testing

- All method names remained the same for both Objective-C and Swift
- Tested build with Cocoapods, Swift Package Manager, and Carthage
- Added additional unit test for error case due to optional unwrapping

### Checklist

- [x] Added a changelog entry

- @sarahkoop @scannillo @jaxdesmarais

### CODE REVIEW

File: Sources/BraintreeAmericanExpress/BTAmericanExpressClient.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    ///   - currencyIsoCode:  The currencyIsoCode to use. Example: 'USD'
+    ///   - completion:  A completion block that is invoked when the request has completed. If the request succeeds,
+    ///   `rewardsBalance` will contain information about the rewards balance and `error` will be `nil` (see exceptions in note);
+    ///   if it fails, `rewardsBalance` will be `nil` and `error` will describe the failure.
+    ///  - Note: If the nonce is associated with an ineligible card or a card with insufficient points, the rewardsBalance will contain this information as `errorMessage` and `errorCode`.
+    public func getRewardsBalance(forNonce nonce: String, currencyIsoCode: String, completion: @escaping (BTAmericanExpressRewardsBalance?, Error?) -> Void) {
+    
+        let parameters = ["currencyIsoCode": currencyIsoCode,
+                          "paymentMethodNonce": nonce]
+        self.apiClient.sendAnalyticsEvent("ios.amex.rewards-balance.start")
```
Comment (sshropshire) [inline Sources/BraintreeAmericanExpress/BTAmericanExpressClient.swift:30]:
```suggestion
        apiClient.sendAnalyticsEvent("ios.amex.rewards-balance.start")
```
Small nit: can we commit `self` where it isn't needed to improve readability?

**Comment by scannillo** [comment]
:shakes-fist: Flakey UI tests. I created another JIRA issue to investigate. #DTBTSDK-1365

File: Sources/BraintreeAmericanExpress/BTAmericanExpressClient.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    ///   - nonce: A nonce representing a card that will be used to look up the rewards balance.
+    ///   - currencyIsoCode:  The currencyIsoCode to use. Example: 'USD'
+    ///   - completion:  A completion block that is invoked when the request has completed. If the request succeeds,
+    ///   `rewardsBalance` will contain information about the rewards balance and `error` will be `nil` (see exceptions in note);
+    ///   if it fails, `rewardsBalance` will be `nil` and `error` will describe the failure.
+    ///  - Note: If the nonce is associated with an ineligible card or a card with insufficient points, the rewardsBalance will contain this information as `errorMessage` and `errorCode`.
+    public func getRewardsBalance(forNonce nonce: String, currencyIsoCode: String, completion: @escaping (BTAmericanExpressRewardsBalance?, Error?) -> Void) {
+    
+        let parameters = ["currencyIsoCode": currencyIsoCode,
+                          "paymentMethodNonce": nonce]
```
Comment (scannillo) [inline Sources/BraintreeAmericanExpress/BTAmericanExpressClient.swift:29]:
```suggestion
        let parameters = ["currencyIsoCode": currencyIsoCode,
                          "paymentMethodNonce": nonce]
```

File: Sources/BraintreeAmericanExpress/BTAmericanExpressClient.swift (hunk 1 above)
Comment (scannillo) [reply Sources/BraintreeAmericanExpress/BTAmericanExpressClient.swift:30]: Good eye - we can remove this in a lot of places

File: Sources/BraintreeAmericanExpress/BTAmericanExpressClient.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        self.apiClient.sendAnalyticsEvent("ios.amex.rewards-balance.start")
+
+        self.apiClient.get("v1/payment_methods/amex_rewards_balance", parameters: parameters) { body, response, error in
+            if let error = error {
```
Comment (scannillo) [inline Sources/BraintreeAmericanExpress/BTAmericanExpressClient.swift:33]: I think we need a capture list here, to avoid retain-cycle case.

**Comment by scannillo** [comment]
I made a JIRA for our follow up task of creating an `ObjC --> Swift Playbook/Guide` here: DTBTSDK-1366.

File: UnitTests/BraintreeAmericanExpressTests/BTAmericanExpressClient_Tests.swift [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import XCTest
 import BraintreeTestShared
-import BraintreeAmericanExpress
+import BraintreeCore
+
```
Comment (scannillo) [inline UnitTests/BraintreeAmericanExpressTests/BTAmericanExpressClient_Tests.swift:4]:
```suggestion
```

**Review by scannillo — APPROVED** [review]
🐧

**Review by sshropshire — APPROVED** [review]
🔥

**Pull request merged**